### PR TITLE
game quality label

### LINKED ISSUE #1478: Fix game quality
image

### PR DESCRIPTION
**Pull request by Geosearchef**
closes #1478

### CODE REVIEW

File: src/main/java/com/faforever/client/replay/ReplayCardController.java [hunk 1]
```diff
@@ -93,7 +93,12 @@ public void setReplay(Replay replay) {
     timeLabel.setText(timeService.asShortTime(replay.getStartTime()));
     modLabel.setText(replay.getFeaturedMod().getDisplayName());
     playerCountLabel.setText(i18n.number(replay.getTeams().values().stream().mapToInt(List::size).sum()));
-    qualityLabel.setText(i18n.get("percentage", (int) ratingService.calculateQuality(replay) * 100));
+    double gameQuality = ratingService.calculateQuality(replay);
+    if (Double.isNaN(gameQuality)) {
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/replay/ReplayCardController.java]: Should it not be if(!Double.isNaN(gameQuality))

File: src/main/java/com/faforever/client/replay/ReplayCardController.java [hunk 2]
```diff
@@ -93,7 +93,12 @@ public void setReplay(Replay replay) {
...
+      qualityLabel.setText(i18n.get("percentage", Math.round(ratingService.calculateQuality(replay) * 100)));
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/replay/ReplayCardController.java]: use gamequality variable

Comment (1-alex98) [reply src/main/java/com/faforever/client/replay/ReplayCardController.java]: (ratingService.calculateQuality(replay) called twice

File: src/main/java/com/faforever/client/replay/ReplayCardController.java (hunk 1 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/replay/ReplayCardController.java]: did you test this at all?

File: src/main/java/com/faforever/client/replay/ReplayDetailController.java [hunk 3]
```diff
@@ -173,7 +173,13 @@ public void setReplay(Replay replay) {
 
     modLabel.setText(replay.getFeaturedMod().getDisplayName());
     playerCountLabel.setText(i18n.number(replay.getTeams().values().stream().mapToInt(List::size).sum()));
-    qualityLabel.setText(i18n.get("percentage", (int) ratingService.calculateQuality(replay) * 100));
+    double gameQuality = ratingService.calculateQuality(replay);
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/replay/ReplayDetailController.java:176]: same here

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

**Comment by 1-alex98** [comment]
as this is a bug caused by changes, maybe a unit test that assures this does not happn ever again

File: src/main/java/com/faforever/client/replay/ReplayCardController.java (hunk 1 above)
Comment (Geosearchef) [reply src/main/java/com/faforever/client/replay/ReplayCardController.java]: Yeah, I was way to sleepy when I did that

File: src/main/java/com/faforever/client/replay/ReplayCardController.java (hunk 2 above)
Comment (Geosearchef) [reply src/main/java/com/faforever/client/replay/ReplayCardController.java]: That was the Plan, don't know what happened to that

File: src/main/java/com/faforever/client/replay/ReplayCardController.java (hunk 1 above)
Comment (Geosearchef) [reply src/main/java/com/faforever/client/replay/ReplayCardController.java]: I tested it after fixing the cast before adding the NaN check, lesson learned, don't change a PR minutes before going to bed.

**Review by Katharsas — COMMENTED** [review]
And when you fix the things rename the commit pls xD

**Comment by Geosearchef** [comment]
Wait what, what the heck did I do?

**Comment by Geosearchef** [comment]
fixed, renamed, unit test added

File: src/main/java/com/faforever/client/replay/ReplayCardController.java [hunk 4]
```diff
@@ -93,7 +93,12 @@ public void setReplay(Replay replay) {
     timeLabel.setText(timeService.asShortTime(replay.getStartTime()));
     modLabel.setText(replay.getFeaturedMod().getDisplayName());
     playerCountLabel.setText(i18n.number(replay.getTeams().values().stream().mapToInt(List::size).sum()));
-    qualityLabel.setText(i18n.get("percentage", (int) ratingService.calculateQuality(replay) * 100));
+    double gameQuality = ratingService.calculateQuality(replay);
+    if (!Double.isNaN(gameQuality)) {
+      qualityLabel.setText(i18n.get("percentage", Math.round(gameQuality * 100)));
+    } else {
+      qualityLabel.setText("-");
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/replay/ReplayCardController.java]: i18n

File: src/main/java/com/faforever/client/replay/ReplayDetailController.java [hunk 5]
```diff
@@ -173,7 +173,13 @@ public void setReplay(Replay replay) {
...
+    if (!Double.isNaN(gameQuality)) {
+      qualityLabel.setText(i18n.get("percentage", Math.round(gameQuality * 100)));
+    } else {
+      qualityLabel.setText("-");
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/replay/ReplayDetailController.java]: i18n

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

**Review by 1-alex98 — DISMISSED** [review]

File: src/main/resources/i18n/messages.properties [hunk 6]
```diff
@@ -711,4 +711,5 @@ linksModerationReport=Moderation Report Form
 linksDiscord=Discord
 linksFacebook=Facebook
 linksDevNews=Dev Blog
-linksYoutube=YouTube
\ No newline at end of file
+linksYoutube=YouTube
+gameQuality.nan=-
```
Comment (Brutus5000) [inline src/main/resources/i18n/messages.properties]: not-a-number is a technical term. But from the user experience side it means "undefined" here. I would not call it `nan` in the translation string (think about the translators not knowing what this is)

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

**Pull request merged**